core/src/main/java/com/scalar/db/transaction/consensuscommit/ConsensusCommitManager.java

```diff
@@ -125,11 +125,13 @@ public ConsensusCommitManager(
 
   // `groupCommitter` must be set before calling this method.
   private CommitHandler createCommitHandler() {
+    // Create CommitHandler(WithGroupCommit) with throwExceptionIfCommittedTransactionExists true
+    // because we want to detect committing a transaction with the duplicated transaction ID.
     if (isGroupCommitEnabled()) {
       return new CommitHandlerWithGroupCommit(
-          storage, coordinator, tableMetadataManager, parallelExecutor, groupCommitter);
+          storage, coordinator, tableMetadataManager, parallelExecutor, groupCommitter, true);
```

Comment:
In the case of the group commit, different transaction IDs are internally created for different groups, even if the same transaction IDs are given from applications. I think it’s still worth enabling this to check the uniqueness of groups, just in case.